src/script/lua_api/l_mapgen.cpp

```diff
@@ -810,6 +810,41 @@ int ModApiMapgen::l_set_mapgen_params(lua_State *L)
 	return 0;
 }
 
+// get_mapgen_edges([mapgen_limit[, chunksize]])
+int ModApiMapgen::l_get_mapgen_edges(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	MapSettingsManager *settingsmgr = getServer(L)->getEmergeManager()->map_settings_mgr;
+
+	// MapSettingsManager::makeMapgenParams cannot be used here because it would
+	// make mapgen settings immutable from then on. Mapgen settings should stay
+	// mutable until after mod loading ends.
+
+	s16 mapgen_limit;
+	if (lua_isnumber(L, 1)) {
+		 mapgen_limit = lua_tointeger(L, 1);
+	} else {
+		std::string mapgen_limit_str;
+		settingsmgr->getMapSetting("mapgen_limit", &mapgen_limit_str);
+		mapgen_limit = stoi(mapgen_limit_str, 0, MAX_MAP_GENERATION_LIMIT);
+	}
+
+	s16 chunksize;
+	if (lua_isnumber(L, 2)) {
+		chunksize = lua_tointeger(L, 2);
+	} else {
+		std::string chunksize_str;
+		settingsmgr->getMapSetting("chunksize", &chunksize_str);
+		chunksize = stoi(chunksize_str, -32768, 32767);
+	}
+
+	std::pair<s16, s16> edges = get_mapgen_edges(mapgen_limit, chunksize);
+	lua_pushinteger(L, edges.first);
+	lua_pushinteger(L, edges.second);
```

Comment:
What would you think about returning a vector here directly? The next action a modder would do is to create one themselves. Furthermore, that would allow to use this function even if there's arbitrary mapgen limits (non-cubic).